cardano-testnet/src/Testnet/Components/DRep.hs

```diff
@@ -210,27 +219,33 @@ data SignedTx
 -- This function takes five parameters:
 --
 -- Returns the generated @File SignedTx In@ file path to the signed transaction file.
-signTx :: (MonadTest m, MonadCatch m, MonadIO m, KeyPair k)
+signTx
+  :: MonadTest m
+  => MonadCatch m
+  => MonadIO m
   => H.ExecConfig -- ^ Specifies the CLI execution configuration.
   -> AnyCardanoEra -- ^ Specifies the current Cardano era.
   -> FilePath -- ^ Base directory path where the signed transaction file will be stored.
   -> String -- ^ Prefix for the output signed transaction file name. The extension will be @.tx@.
   -> File TxBody In -- ^ Transaction body to be signed, obtained using 'createCertificatePublicationTxBody' or similar.
-  -> [k] -- ^ List of payment key pairs used for signing the transaction.
+  -> [SomeKeyPair] -- ^ List of payment key pairs used for signing the transaction.
```

Comment:
The comment needs to be modified to show that we can sign with other keys (e.g stake keys)